src/betterproto/plugin/models.py

```diff
@@ -0,0 +1,713 @@
+
+
+@dataclass
+class OneOfFieldCompiler(FieldCompiler):
+    @property
+    def betterproto_field_args(self) -> "str":
+        args = super().betterproto_field_args
+        group = self.parent.proto_obj.oneof_decl[self.proto_obj.oneof_index].name
+        args = args + f', group="{group}"'
+        return args
+
+
+@dataclass
+class MapEntryCompiler(FieldCompiler):
+    py_k_type: Type = PLACEHOLDER
+    py_v_type: Type = PLACEHOLDER
+    proto_k_type: str = PLACEHOLDER
+    proto_v_type: str = PLACEHOLDER
+
+    def __post_init__(self):
+        """Explore nested types and set k_type and v_type if unset."""
+        map_entry = f"{self.proto_obj.name.replace('_', '').lower()}entry"
+        for nested in self.parent.proto_obj.nested_type:
+            if nested.name.replace("_", "").lower() == map_entry:
+                if nested.options.map_entry:
+                    # Get Python types
+                    self.py_k_type = FieldCompiler(
+                        parent=self, proto_obj=nested.field[0],  # key
+                    ).py_type
+                    self.py_v_type = FieldCompiler(
+                        parent=self, proto_obj=nested.field[1],  # key
+                    ).py_type
+                    # Get proto types
+                    self.proto_k_type = self.proto_obj.Type.Name(nested.field[0].type)
+                    self.proto_v_type = self.proto_obj.Type.Name(nested.field[1].type)
+        super().__post_init__()  # call FieldCompiler-> MessageCompiler __post_init__
+
+    def get_field_string(self, indent: int = 4) -> str:
+        """Construct string representation of this field."""
+        name = f"{self.py_name}"
+        annotations = f": {self.annotation}"
+        betterproto_field_type = (
+            f"betterproto.map_field("
+            f"{self.proto_obj.number}, betterproto.{self.proto_k_type}, "
+            f"betterproto.{self.proto_v_type})"
+        )
+        return name + annotations + " = " + betterproto_field_type
+
+    @property
+    def annotation(self):
+        return f"Dict[{self.py_k_type}, {self.py_v_type}]"
+
+    @property
+    def repeated(self):
+        return False  # maps cannot be repeated
+
+
+@dataclass
+class EnumDefinitionCompiler(MessageCompiler):
+    """Representation of a proto Enum definition."""
+
+    proto_obj: EnumDescriptorProto = PLACEHOLDER
+    entries: List["EnumDefinitionCompiler.EnumEntry"] = PLACEHOLDER
+
+    @dataclass(unsafe_hash=True)
+    class EnumEntry:
+        """Representation of an Enum entry."""
+
+        name: str
+        value: int
+        comment: str
+
+    def __post_init__(self):
+        # Get entries
+        self.entries = [
+            self.EnumEntry(
+                name=v.name,
+                value=v.number,
+                comment=get_comment(
+                    proto_file=self.proto_file, path=self.path + [2, i]
+                ),
+            )
+            for i, v in enumerate(self.proto_obj.value)
```

Comment:
A more descriptive variable name could help to make clear what elements are iterated over. Are they Enum entries, EnumEntryDescriptors, or fielddescriptors?

Reply:
Agreed. I tried to clarify, let me know if this helps!